dot/digest/digest.go

```diff
@@ -201,17 +201,26 @@ func (h *Handler) handleBabeConsensusDigest(digest scale.VaryingDataType, header
 	case types.BABEOnDisabled:
 		return nil
 
-	case types.NextConfigData:
-		currEpoch, err := h.epochState.GetEpochForBlock(header)
+	case types.VersionedNextConfigData:
+		nextVersionedConfigData := digestValue.(types.VersionedNextConfigData)
+		nextConfigDataVersion, err := nextVersionedConfigData.Value()
 		if err != nil {
-			return fmt.Errorf("cannot get epoch for block %d (%s): %w",
-				header.Number, headerHash, err)
+			return fmt.Errorf("getting digest value: %w", err)
 		}
 
-		nextEpoch := currEpoch + 1
-		h.epochState.StoreBABENextConfigData(nextEpoch, headerHash, val)
-		h.logger.Debugf("stored BABENextConfigData data: %v for hash: %s to epoch: %d", digest, headerHash, nextEpoch)
-		return nil
+		switch nextConfigData := nextConfigDataVersion.(type) {
+		case types.NextConfigDataV1:
+			currEpoch, err := h.epochState.GetEpochForBlock(header)
+			if err != nil {
+				return fmt.Errorf("cannot get epoch for block %d (%s): %w", header.Number, headerHash, err)
```

Comment:
https://github.com/ChainSafe/gossamer/pull/3228#discussion_r1181505625
```suggestion
				return fmt.Errorf("get epoch for block %d (%s): %w", header.Number, headerHash, err)
```

Reply:
Done! thank you